pyanaconda/core/util.py

```diff
@@ -165,6 +167,68 @@ def preexec():
     return partsubp(preexec_fn=preexec)
 
 
+class WaylandStatus:
+    """Status of Wayland launch.
+
+    Values of an instance can be modified from the handler functions.
+    """
+    def __init__(self):
+        self.started = False
+        self.timed_out = False
+
+    def needs_waiting(self):
+        return not (self.started or self.timed_out)
+
+
+def startWl(weston_config=WESTON_CONFIG, output_redirect=None, timeout=WAYLAND_TIMEOUT):
+    """ Start Weston for Wayland and return once Weston is ready to accept connections.
+
+        We can identify whether Weston is ready by testing if
+        the Wayland socket is open yet. Once it is, we can return success.
+
+        :param weston_config: The weston.ini(5) configuration to use, as a dictionary
+        :param output_redirect: file or file descriptor to redirect stdout and stderr to
+        :param timeout: Number of seconds to timing out.
+    """
+    wl_status = WaylandStatus()
+
+    # Create the config file for Weston
+    weston_config_file = tempfile.NamedTemporaryFile(suffix="-wl-weston-sysinstall-ini",
+                                                     delete=False)
+    weston_config_ini = configparser.ConfigParser()
+    for section, options in weston_config.items():
+        weston_config_ini.add_section(section)
+        for key, value in options.items():
+            weston_config_ini.set(section, key, str(value))
+    weston_config_ini.write(weston_config_file)
+
+    log.debug("Starting Weston.")
+    argv = ["weston", f"--config={weston_config_file}", "--log=/tmp/weston.log",
+            f"--socket={WAYLAND_SOCKET_NAME}"]
```

Comment:
We aren't using f-strings anywhere else.

Reply:
f-strings were added in Python 3.6. I think it's pretty reasonable to state that we're probably not going to be running this Anaconda codebase on something older than that, since Python 3.6 is what RHEL 8 shipped with. I can certainly change it, but I liked how it make this easier to read.

Reply:
Sure, it would be possible to use f-strings on the platforms we care about, it has been more a question of team policy or consistency so far not to use them in the Anaconda code.